### ISSUE TITLE
REPL editing broken in $(julia -e 'import GAP; print(GAP.gap_exe())')

### ISSUE THREAD

**Issue by mohamed-barakat** [oscar-system/GAP.jl#415]
The backspace or the arrows corrupt the input line.

**Comment by fingolfin** [comment]
Indeed, I can reproduce this. I am linking against a copy of GNU readline produced by @benlorenz for Polymake.jl, see https://github.com/benlorenz/readlineBuilder -- the idea being that I want to use the identical dylibs as they are, to minimize the risk of conflicts (and soon we'll both want to switch to the corresponding _Jll, that'll be a relief)

**Comment by benlorenz** [comment]
We could switch to the binaries from the jll without switching to the jll more or less right now by using this [build_Readline.v8.0.0.jl.txt](https://github.com/oscar-system/GAP.jl/files/4480591/build_Readline.v8.0.0.jl.txt) generated from the Yggdrasil `generate_build.jl`. If this helps your issue I can do the switch for polymake.

polymake doesn't really need readline when it is used in library mode, but it is required as a dependency. When we tried it with the normal polymake command line interface we needed some TERMINFO environment variable as readline is not properly relocatable and might not find the correct database:
```
export TERMINFO=/path/to/Polymake.jl/deps/usr/share/terminfo
```

**Comment by mohamed-barakat** [comment]
This issue should be reopened or another similar one should be opened.

**Comment by fingolfin** [comment]
That was an accident: I intentionally modified the "fixes BLA" comment in PR #427, but that wasn't enough (probably because of the still relatively GitHub feature of "linking" issues to PRs)

**Comment by fingolfin** [comment]
@mohamed-barakat this should be resolved in GAP 0.4.0, can you please check?

**Comment by mohamed-barakat** [comment]
Yes, resolved :) Thank you for the excellent work.
